### PR TITLE
Add a default for maximum products generated by a reaction (maxProduc…

### PR DESCRIPTION
**Pull request by bp-kelley**
…ts=1000)

#### Reference Issue

Reagents that have a ridiculous number of matches can play havoc with the reaction runner.  This limits the possible number of products to a sane amount.

### CODE REVIEW

File: Code/GraphMol/ChemReactions/Wrap/testReactionWrapper.py [hunk 1]
```diff
@@ -694,5 +694,21 @@ def testProperties(self):
     self.assertTrue(nrxn.HasProp("intprop"))
     self.assertEquals(nrxn.GetIntProp("intprop"),3)
 
+  def testRoundTripException(self):
```
Comment (greglandrum) [inline Code/GraphMol/ChemReactions/Wrap/testReactionWrapper.py]: This test doesn't seem related to the PR. Are you sure you meant to include it?
(having more tests is always good, I just want to check)

Comment (bp-kelley) [reply Code/GraphMol/ChemReactions/Wrap/testReactionWrapper.py]: It was related to #1985 like the one in SetQuery, they caused exceptions.  It isn't related to this PR but it isn't a terrible test.  I can remove it if you like

Comment (greglandrum) [reply Code/GraphMol/ChemReactions/Wrap/testReactionWrapper.py]: Nope, I just wanted to be sure that it didn't accidentally creep in.
